From abae68f4e28ab4ce6c08cf938aad340419bb7af7 Mon Sep 17 00:00:00 2001
From: Julien Gilli <julien.gilli@joyent.com>
Date: Thu, 23 Nov 2017 17:25:36 -0800
Subject: [PATCH] VOLAPI-94 sdcadm experimental nfs-volumes commands error on
 some core services using release images Reviewed by: Trent Mick
 <trentm@gmail.com> Approved by: Trent Mick <trentm@gmail.com>

---
 lib/cli/do_nfs_volumes.js | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/lib/cli/do_nfs_volumes.js b/lib/cli/do_nfs_volumes.js
index fcbd9cd..75afd09 100644
--- a/lib/cli/do_nfs_volumes.js
+++ b/lib/cli/do_nfs_volumes.js
@@ -230,6 +230,11 @@ function do_nfs_volumes(subcmd, opts, args, cb) {
                     branch: /^master$/,
                     timestamp: '20171110T184239Z',
                     commit: 'a60a380'
+                },
+                {
+                    branch: /^release-\d{8}$/,
+                    timestamp: '20171123T073154Z',
+                    commit: '04b9561'
                 }
             ],
             volapi: [
@@ -264,6 +269,11 @@ function do_nfs_volumes(subcmd, opts, args, cb) {
                     branch: /^master$/,
                     timestamp: '20171110T202313Z',
                     commit: '9d39e1b'
+                },
+                {
+                    branch: /^release-\d{8}$/,
+                    timestamp: '20171123T073050Z',
+                    commit: '8c37b7e'
                 }
             ];
         } else {
@@ -272,6 +282,11 @@ function do_nfs_volumes(subcmd, opts, args, cb) {
                     branch: /^master$/,
                     timestamp: '20171110T191649Z',
                     commit: '788a08f'
+                },
+                {
+                    branch: /^release-\d{8}$/,
+                    timestamp: '20171123T072329Z',
+                    commit: 'aa9022d'
                 }
             ];
         }
@@ -425,6 +440,22 @@ function do_nfs_volumes(subcmd, opts, args, cb) {
 
                     assert.string(vmUuid, 'vmUuid');
 
+                    if (!/^release-\d{8}$/.test(versionBranch) &&
+                        versionBranch !== 'master') {
+                        /*
+                         * This image was built from a feature branch and was
+                         * most likely installed from the experimental channel,
+                         * so there's no way to really know whether it supports
+                         * NFS volumes or not. We'll consider that the operator
+                         * knows what they're doing, and just output a warning.
+                         */
+                        self.progress('Image for service %s at version %s is ' +
+                            'not from a release or master branch, it may not ' +
+                            'support NFS volumes', serviceName,
+                            versionInfo.version);
+                            return;
+                    }
+
                     if (!versionSeemsGreaterOrEqualThan({
                         branch: versionBranch,
                         timestamp: versionTimestamp,
-- 
2.21.0

